mgmt: mcumgr: Add Kconfig option to manage SMP service registration

mkapala-nordic opened the pull request:
Added the CONFIG_MCUMGR_TRANSPORT_BT_DYNAMIC_SVC_REGISTRATION Kconfig option to manage how the SMP service should be registered. By default the SMP service must be registered at runtime. If this Kconfig option is disabled, the SMP service is statically defined and registered.

This change allows to opt out of using the CONFIG_BT_GATT_DYNAMIC_DB Kconfig option and as a result, lower the memory usage.

CC: @kapi-no @MarekPieta @zycz

mkapala-nordic committed:
mgmt: mcumgr: Add Kconfig option to manage SMP service registration

Added the CONFIG_MCUMGR_TRANSPORT_BT_DYNAMIC_SVC_REGISTRATION Kconfig
option to manage how the SMP service should be registered.
By default the SMP service must be registered at runtime.
If this Kconfig option is disabled, the SMP service is statically
defined and registered.

This change allows to opt out of using the CONFIG_BT_GATT_DYNAMIC_DB
Kconfig option and as a result, lower the memory usage.

Signed-off-by: Mateusz Kapala <mateusz.kapala@nordicsemi.no>

nordicjm commented on subsys/mgmt/mcumgr/transport/Kconfig.bluetooth:
```diff
@@ -92,4 +91,12 @@ config MCUMGR_TRANSPORT_BT_CONN_PARAM_CONTROL_RETRY_TIME
+	select BT_GATT_DYNAMIC_DB
+	default y
+	help
+	  When enabled, the SMP service must be registred at runtime.
```
*registered

MarekPieta replied:
`is automatically registered during boot and then can be dynamically registered/unregistered using dedicated API.`
(application actually dose not need to register the service on boot)

nordicjm reviewed (changes requested):
Changes are OK. Compliance needs fixing, replace the 8-space blocks with tabs

kapi-no commented:
Please add the CI build configuration for the non-default value of the `MCUMGR_TRANSPORT_BT_DYNAMIC_SVC_REGISTRATION` Kconfig:

`CONFIG_MCUMGR_TRANSPORT_BT_DYNAMIC_SVC_REGISTRATION=n`

It should ensure that both variants compile correctly.

(SMP SVR sample should be a proper place to add this configuration.)

mkapala-nordic committed:
samples: subsys: mgmt: mcumgr: smp_svr: Add BT static svc registration

Added build configuration for the
CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP_WITH_REVERT=n (non-default
option) to test building the smp_svr sample with statically defined
and registered SMP service.

Signed-off-by: Mateusz Kapala <mateusz.kapala@nordicsemi.no>
